src/decoder.ts

```diff
@@ -383,6 +382,72 @@ export class Decoder<A> {
     });
   }
 
+  /**
+   * Decoder for fixed-length arrays, aka Tuples.
+   *
+   * Supports up to 8-tuples.
+   *
+   * Example:
+   * ```
+   * tuple([number(), number(), string()]).run([5, 10, 'px'])
+   * // => {ok: true, result: [5, 10, 'px']}
+   * ```
+   */
+  static tuple<A>(decoder: [Decoder<A>]): Decoder<[A]>;
+  static tuple<A, B>(decoder: [Decoder<A>, Decoder<B>]): Decoder<[A, B]>;
+  static tuple<A, B, C>(decoder: [Decoder<A>, Decoder<B>, Decoder<C>]): Decoder<[A, B, C]>;
+  static tuple<A, B, C, D>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>]
+  ): Decoder<[A, B, C, D]>;
+  static tuple<A, B, C, D, E>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>]
+  ): Decoder<[A, B, C, D, E]>;
+  static tuple<A, B, C, D, E, F>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>]
+  ): Decoder<[A, B, C, D, E, F]>;
+  static tuple<A, B, C, D, E, F, G>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>, Decoder<G>]
+  ): Decoder<[A, B, C, D, E, F, G]>;
+  static tuple<A, B, C, D, E, F, G, H>(
+    decoder: [
+      Decoder<A>,
+      Decoder<B>,
+      Decoder<C>,
+      Decoder<D>,
+      Decoder<E>,
+      Decoder<F>,
+      Decoder<G>,
+      Decoder<H>
+    ]
+  ): Decoder<[A, B, C, D, E, F, G, H]>;
+  static tuple<A>(decoders?: Decoder<A>[]) {
+    return new Decoder((json: unknown) => {
+      if (isJsonArray(json) && decoders) {
+        if (json.length !== decoders.length) {
+          return Result.err({
+            message: `expected an array of length ${decoders.length}, got an array of length ${
```

Comment:
nit: I'm tempted to make this "expected a tuple of length ${decoders.length}"
I say that because one issue with the library right now is that it can be a bit difficult to figure out which decoder failed, if you aggregate your error messages. At least one other similar library solves that by allowing you to pass a name param to each decoder you make. We don't have that right now, so anything that better hints at what combinator failed is good.

Reply:
Good idea! Updated in 247b86b.